### ISSUE TITLE
Clipping performance implications (question)

### ISSUE THREAD

**Issue by jmlee2k** [pixijs/pixi-spine#190]
Hi, thanks for adding clipping support, it works great!

I'm just wondering about the performance implications (if there are any).  For example, is using a 4-vertex non-animated rectangle for clipping comparable to masking the entire PIXI.spine.Spine object in code?

**Comment by ivanpopelyshev** [comment]
Clipping in pixi-spine animations always uses stencil buffer. I didnt make benchmarks regarding that. It may be that on mobile devices standart spine-ts approach of manual clipping quads is better sometimes. 

In pixi-spine clipping is affecting GPU , while in spine-ts it relies only on CPU. In any case, it breaks batches and you really dont want to place 100 objects using clipping on screen.

I may optimize it that way if there are only four points it will use SCISSOR instead, and scissor is really fast.

**Comment by jmlee2k** [comment]
Thanks for the thorough answer.  We'll only have a max of about 25 objects using clipping at once, and in practice, it will be much lower.  We'll see how it performs on mobile devices.

Just out of curiosity, in the case of a 4-vertex polygon would you optimize it by using a Rect instead of a Polygon in Spine.prototype.createGraphics/Spine.prototype.updateGraphics?

**Comment by ivanpopelyshev** [comment]
the problem is that i have to detect it in updateGraphics and change polygon to rect instead, that way pixi sees that it can use scissor instead.

**Comment by jmlee2k** [comment]
Makes sense.  Thanks for the info!  Do you have any formal way to vote for or express interest in a feature?

**Comment by ivanpopelyshev** [comment]
No, its random, do you have any idea of how to do that across all pixi plugins?

I'm taking commissions for small money to make certain features very fast when people need them :) Right now I'm taking a commission on https://github.com/gameofbombs/pixi-super-atlas , that will allow to share the atlas between multiple spine objects. Even more, if packed png is small, you can use some packer (webpack or another) and then super-atlas will process all the small images to fit inside one atlas that'll be uploaded to videomemory.

**Comment by jmlee2k** [comment]
Nope, I'm not aware of anywhere to vote for features for the entire pixi ecosystem.

That super atlas sounds really interesting.  If you're interested in developing the scissor test optimization on a commission, let me know, and we can talk separately.  I'd happily contribute a PR for it, but it's a bit over my head. :)

**Comment by ivanpopelyshev** [comment]
Are you sure that you need it? Did you make tests on certain devices with masks and without them?

**Comment by jmlee2k** [comment]
No, not yet, I'll let you know how it goes.

**Comment by jmlee2k** [comment]
Looks like the performance is fine for our use case.  Obviously using the scissor test would be better since it is a rect, but this isn't a blocker for us.
